src/python/pants/util/collections_test.py

```diff
@@ -85,3 +86,23 @@ def test_ensure_str_list() -> None:
         ensure_str_list(0)  # type: ignore[arg-type]
     with pytest.raises(ValueError):
         ensure_str_list([0, 1])  # type: ignore[list-item]
+
+
+@pytest.mark.parametrize("size_min", [0, 1, 16, 32, 64, 128])
+def test_partition_sequentially(size_min: int) -> None:
+    # Adding an item at any position in the input sequence should affect either 1 or 2 (if the added
+    # item becomes a boundary) buckets in the output.
+
+    def partitioned_buckets(items: list[str]) -> set[tuple[str, ...]]:
+        return set(tuple(p) for p in partition_sequentially(items, key=str, size_min=size_min))
+
+    # We start with base items containing every other element from a sorted sequence.
+    all_items = sorted((f"item{i}" for i in range(0, 64)))
```

Comment:
Please do not burn a tree for this, but FYI you already had a generator expression and the extra `()` is superfluous:

```suggestion
    all_items = sorted(f"item{i}" for i in range(0, 64))
```

Reply:
@Eric-Arellano I think there is a flake8 plugin that detects those... maybe we should add it to this repo.
https://pypi.org/project/flake8-comprehensions/

Reply:
That'd be great!

Reply:
ok. will work on this. PRs coming soon.